src/hotspot/share/runtime/thread.hpp

```diff
@@ -1206,97 +1136,24 @@ class JavaThread: public Thread {
   }
 
   // Suspend/resume support for JavaThread
- private:
-  inline void set_ext_suspended();
-  inline void clear_ext_suspended();
-
- public:
-  void java_suspend(); // higher-level suspension logic called by the public APIs
-  void java_resume();  // higher-level resume logic called by the public APIs
-  int  java_suspend_self(); // low-level self-suspension mechanics
-
-  // Synchronize with another thread that is deoptimizing objects of the
-  // current thread, i.e. reverts optimizations based on escape analysis.
-  void wait_for_object_deoptimization();
-
- private:
-  // mid-level wrapper around java_suspend_self to set up correct state and
-  // check for a pending safepoint at the end
-  void java_suspend_self_with_safepoint_check();
+  bool java_suspend(); // higher-level suspension logic called by the public APIs
+  bool java_resume();  // higher-level resume logic called by the public APIs
+  bool is_suspended()                 { return _handshake.is_suspended(); }
+  bool suspend_request_pending()      { return _handshake.suspend_request_pending(); }
```

Comment:
If you use is_suspended() to check for suspension in objectMonitor::enter() then we can remove this method.

Reply:
Fixed